### ISSUE TITLE
IPAD IOS 9.3.4 Bug

### ISSUE THREAD

**Issue by TaiVuong** [libgdx/libgdx#4257]
I just update my ipad to 9.3.4 and the old build force stop.

when i try to build the new --> LOCKDOWN_E_INVALID_HOST_ID

libgdx : 1.9.3 , 1.9.4-SNAPSHOT
robovm : 2.1.0, 2.2.0-SNAPSHOT

others libs: freetype, gdx-pay, joda-time:2.9.4

_ eclipse plugin: 
+ http://robovm.mobidevelop.com/downloads/snapshots/eclipse/
+ http://robovm.mobidevelop.com/downloads/releases/eclipse/2.1.0/site/

OSX version: 10.11.6
```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':ios:launchIOSDevice'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
	at org.gradle.tooling.internal.provider.runner.BuildModelActionRunner.run(BuildModelActionRunner.java:46)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.tooling.internal.provider.runner.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:58)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:48)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:52)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:47)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:66)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: org.gradle.api.GradleException: Failed to launch IOS Device
	at org.robovm.gradle.tasks.IOSDeviceTask.invoke(IOSDeviceTask.java:48)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 70 more
Caused by: org.robovm.libimobiledevice.LibIMobileDeviceException: LOCKDOWN_E_INVALID_HOST_ID
	at org.robovm.libimobiledevice.LockdowndClient.checkResult(LockdowndClient.java:151)
	at org.robovm.libimobiledevice.LockdowndClient.<init>(LockdowndClient.java:59)
	at org.robovm.libimobiledevice.util.AppLauncher.install(AppLauncher.java:545)
	at org.robovm.compiler.target.ios.AppLauncherProcess.execAsync(AppLauncherProcess.java:62)
	at org.robovm.compiler.target.AbstractTarget.doLaunch(AbstractTarget.java:490)
	at org.robovm.compiler.target.ios.IOSTarget.doLaunch(IOSTarget.java:594)
	at org.robovm.compiler.target.AbstractTarget.launch(AbstractTarget.java:486)
	at org.robovm.compiler.AppCompiler.launchAsync(AppCompiler.java:828)
	at org.robovm.compiler.AppCompiler.launch(AppCompiler.java:813)
	at org.robovm.compiler.AppCompiler.launch(AppCompiler.java:808)
	at org.robovm.gradle.tasks.IOSDeviceTask.invoke(IOSDeviceTask.java:46)
	... 78 more
```

**Comment by mustii82** [comment]
try the second iOS Backend(MultiOS) There is no more support for the robovm backend since the company closed

**Comment by intrigus** [comment]
@mustii82 That's not correct.
There is still support for the oss robovm backend.

**Comment by TaiVuong** [comment]
The last update of Robovm 2.2.0 solved the problem.
Thank you all.

**Comment by moksalahti** [comment]
@TaiVuong I'm having the same problems but cannot fix it using 2.2.0. Do you mean the version 2.2.0 from com.mobidevelop.robovm (that's what I tried)? The biggest version I can find from org.robovm is 1.14.1. I'm using robovm-maven-plugin to build the app. I guess you are using gradle?

**Comment by TaiVuong** [comment]
@moksalahti,  use version 2.2.0 from com.mobidevelop.robovm, update eclipse plugin, clear cache.
I use both eclipse plugin + gradle.
If it does't work then try to restart your OS.

**Comment by guojinshang** [comment]
I have the same problem on my ipad mini 2.

**Comment by guojinshang** [comment]
I'm still using the org.robovm now because the 2.2.0 did not support robopod fully now.

**Comment by moksalahti** [comment]
@TaiVuong I tried with 2.2.0 (com.mobidevelop.robovm) and robovm-maven-plugin 1.14.0 (org.robovm) but I still get error LOCKDOWN_E_PAIRING_DIALOG_RESPONSE_PENDING on the first attempt to build. At the same time on my iPad I get the "Trust this computer" question, to which I answer "Trust". After that I get error LOCKDOWN_E_INVALID_HOST_ID for all build attempts I try.

I've tried resetting all settings on the iPad, restarting it but nothing seems to help. Here's the stack trace for the error:

`[ERROR] Failed to execute goal org.robovm:robovm-maven-plugin:1.14.0:ios-device (default-cli) on project myapp-robovm: Failed to launch IOS Device: LOCKDOWN_E_INVALID_HOST_ID -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.robovm:robovm-maven-plugin:1.14.0:ios-device (default-cli) on project myapp-robovm: Failed to launch IOS Device
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to launch IOS Device
	at org.robovm.maven.plugin.IOSDeviceMojo.execute(IOSDeviceMojo.java:54)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: org.robovm.libimobiledevice.LibIMobileDeviceException: LOCKDOWN_E_INVALID_HOST_ID
	at org.robovm.libimobiledevice.LockdowndClient.checkResult(LockdowndClient.java:124)
	at org.robovm.libimobiledevice.LockdowndClient.<init>(LockdowndClient.java:48)
	at org.robovm.compiler.target.ios.AppleDeviceTarget.getDeviceInfo(AppleDeviceTarget.java:424)
	at org.robovm.compiler.target.ios.AppleDeviceTarget.isDeviceCompatible(AppleDeviceTarget.java:439)
	at org.robovm.compiler.target.ios.AppleDeviceTarget.createDeviceLauncher(AppleDeviceTarget.java:281)
	at org.robovm.compiler.target.ios.AppleDeviceTarget.createLauncher(AppleDeviceTarget.java:165)
	at org.robovm.compiler.target.AbstractTarget.doLaunch(AbstractTarget.java:499)
	at org.robovm.compiler.target.ios.AppleDeviceTarget.doLaunch(AppleDeviceTarget.java:762)
	at org.robovm.compiler.target.AbstractTarget.launch(AbstractTarget.java:495)
	at org.robovm.compiler.AppCompiler.launchAsync(AppCompiler.java:792)
	at org.robovm.compiler.AppCompiler.launch(AppCompiler.java:777)
	at org.robovm.compiler.AppCompiler.launch(AppCompiler.java:772)
	at org.robovm.maven.plugin.IOSDeviceMojo.execute(IOSDeviceMojo.java:51)
	... 21 more
`

**Comment by TaiVuong** [comment]
try update OSX.
update  Xcode.
delete project/ios/build

gradle 
```
...
classpath 'com.mobidevelop.robovm:robovm-gradle-plugin:2.2.0'
...
allprojects {
    apply plugin: "eclipse"
    apply plugin: "idea"

    version = '1.0'
    ext {
        gdxVersion = '1.9.4'
        roboVMVersion = '2.2.0'
    }

    repositories {
        mavenLocal()
        mavenCentral()
        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
        maven { url "https://oss.sonatype.org/content/repositories/releases/" }
    }
}
...
project(":ios") {
    apply plugin: "java"
    apply plugin: "robovm"
    dependencies {
        compile project(":core")
        compile "com.mobidevelop.robovm:robovm-rt:$roboVMVersion"
        compile "com.mobidevelop.robovm:robovm-cocoatouch:$roboVMVersion"
        compile "com.badlogicgames.gdx:gdx-backend-robovm:$gdxVersion"
        compile "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-ios"
        compile "com.badlogicgames.gdx:gdx-freetype-platform:$gdxVersion:natives-ios"
        compile "com.mobidevelop.robovm:robopods-google-mobile-ads-ios:2.2.0-SNAPSHOT"
        
    }
}
```
build with gradle.

**Comment by moksalahti** [comment]
@TaiVuong Thanks for the suggestions. I'll see if moving to gradle is feasible (this is quite a large project I inherited from a previous developer...) Here's how I managed to work around the problem (this might also benefit you, @guojinshang): Instead of installing the app directly to the device using Maven build command:

`mvn -Probovm clean compile robovm:ios-device`

I build the IPA package with Maven:

`mvn -Probovm clean compile robovm:create-ipa`

and then install it to the device using XCode. It's one extra step when trying the developed version on the device but it only takes a minute so it's not a big deal.

**Comment by HarryFx** [comment]
We face the same issue with RoboVM 1.1.3 getting: 
 org.robovm.libimobiledevice.LibIMobileDeviceException: LOCKDOWN_E_INVALID_HOST_ID

This also affect the current Gluon plugin (which uses RoboVM 1.8.0). Tried everything. Can still deploy directly from XCode but not using RoboVM. Multi-OS Engine deployment works as well.

I can restore OS X from an older backup and than it works again. It's the 2nd time it happens within 3 weeks.

I ended up installing El Capitain 10.11.6 in a VM (Fusion from vmware, Virtualbox is too slow for an OSX guest), setup my dev environment (IntelliJ) and connect the iOS device straight into the VM and I can deploy from IntelliJ again (using the same signing identities and provisioning profiles). Now I can make VM snapshots and it will never break again.

The first time the issue occurred I was testing XCode 8 Beta to see if RoboVM still deploys to iOS 10 devices (it didn't). Then I restored my OSX and all was good. Now three weeks later the same. Haven't touched XCode at all. I suspect while evaluating Gluon and MOE something happened to libmobiledevice (it's used by all dev tools).

What a pita, thanks again Microsoft!

**Comment by HarryFx** [comment]
Found the underlying issue. **iTunes update breaks RoboVM 1.13.0**. Last know iTunes version which works is **iTunes 12.3.3.17** (clean El Capitan 10.11.6 install). More details here: [LibGDX](http://badlogicgames.com/forum/viewtopic.php?t=23642&p=96851#p96881)

**Comment by HarryFx** [comment]
Issue fixed, **RoboVM 1.13.0 works again**! Details here [LibGDX](http://badlogicgames.com/forum/viewtopic.php?f=11&t=23642&p=96881#p96908)

**Comment by moksalahti** [comment]
@HarryFx Great news. Thanks for letting us know!
